sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/aio/_client.py

```diff
@@ -61,34 +61,35 @@ def __init__(self, key: "Union[KeyVaultKey, str]", credential: "AsyncTokenCreden
         self._key_id = None  # type: Optional[KeyVaultResourceId]
 
         if isinstance(key, KeyVaultKey):
-            self._key = key.key
+            self._key = key.key  # type: Union[JsonWebKey, KeyVaultKey, str, None]
             self._key_id = parse_key_vault_id(key.id)
             if key.properties._attributes:  # pylint:disable=protected-access
                 self._not_before = key.properties.not_before
                 self._expires_on = key.properties.expires_on
         elif isinstance(key, str):
             self._key = None
             self._key_id = parse_key_vault_id(key)
-            self._keys_get_forbidden = None  # type: Optional[bool]
+            self._keys_get_forbidden = False
         elif self._jwk:
             self._key = key
         else:
             raise ValueError("'key' must be a KeyVaultKey instance or a key ID string including a version")
 
-        if not (self._jwk or self._key_id.version):
+        if not (self._jwk or self._key_id.version if self._key_id else None):
             raise ValueError("'key' must include a version")
 
+        self._initialized = None  # type: Optional[bool]
         if self._jwk:
             try:
-                self._local_provider = get_local_cryptography_provider(self._key)
+                self._local_provider = get_local_cryptography_provider(cast(JsonWebKey, self._key))
                 self._initialized = True
             except Exception as ex:  # pylint:disable=broad-except
                 raise ValueError("The provided jwk is not valid for local cryptography") from ex
         else:
             self._local_provider = NoLocalCryptography()
             self._initialized = False
 
-        self._vault_url = None if self._jwk else self._key_id.vault_url
+        self._vault_url = None if self._jwk else self._key_id.vault_url  # type: ignore
```

Comment:
This line should match the sync client's, and other comments I made apply here (just emphasizing since I've forgotten to carry changes over to async clients before 😅)

Reply:
Good reminder!